src/main/scala/ai/datahunters/md/reader/SolrHashReader.scala

```diff
@@ -0,0 +1,102 @@
+package ai.datahunters.md.reader
+
+import java.util
+
+import ai.datahunters.md.config.writer.SolrWriterConfig
+import ai.datahunters.md.processor.ColumnNamesConverterFactory
+import ai.datahunters.md.util.SolrClientBuilder
+import org.apache.solr.client.solrj.impl.CloudSolrClient
+import org.apache.solr.common.params.MapSolrParams
+import org.apache.solr.common.{SolrDocument, SolrDocumentList}
+import org.apache.spark.sql.types.{DataTypes, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+
+import scala.collection.JavaConversions._
+
+/**
+  * Read  id and hashes to DataFrame mainly for processing reduction
+  *
+  * @param sparkSession
+  * @param config
+  */
+case class SolrHashReader(sparkSession: SparkSession,
```

Comment:
@dawidsau Please add some note that this is experimental due to loading all IDs from Solr into driver which could cause OOM in case of large collections